generator/feature_segments_checker/feature_segments_checker.cpp

```diff
@@ -75,14 +79,47 @@ void WriteCSV(Cont const & cont, string const & fileName)
     fout << a.first << "," << a.second << endl;
 }
 
-/// \returns expected point altitude in meters according to linear model.
-double ExpectedPointAltitude(int16_t startAltitudeMeters, int16_t endAltitudeMeters,
-                             double distFromStartMeters, double featureLengthMeters)
+/// \returns y = k * x + b. It's an expected altitude in meters.
+double ExpectedPointAltitude(double k, double b, double distFromStartMeters)
 {
-  if (featureLengthMeters == 0.0)
-    return 0;
-  double const k = (endAltitudeMeters - startAltitudeMeters) / featureLengthMeters;
-  return startAltitudeMeters + k * distFromStartMeters;
+  return k * distFromStartMeters + b;
+}
+
+template <class T>
+long double SumWithDevisor(vector<T> const & vec, long double devisor)
```

Comment:
Why not to create a method, say, Sum(), that will be a nice and readable shortcut for accumulate, and everywhere where you need to divide the sum of some vector by some value, directly divide them?